pkg/ext-proc/backend/inferencepool_reconciler.go

```diff
@@ -29,26 +29,31 @@ type LLMServerPoolReconciler struct {
 	Zone           string
 }
 
-func (c *LLMServerPoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+func (c *InferencePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	if req.NamespacedName.Name != c.ServerPoolName && req.NamespacedName.Namespace != c.Namespace {
 		return ctrl.Result{}, nil
 	}
-	klog.V(2).Infof("Reconciling LLMServerPool %v", req.NamespacedName)
+	klog.V(1).Info("reconciling InferencePool", req.NamespacedName)
 
-	serverPool := &v1alpha1.LLMServerPool{}
+	serverPool := &v1alpha1.InferencePool{}
 	if err := c.Get(ctx, req.NamespacedName, serverPool); err != nil {
-		klog.Errorf("Unable to get LLMServerPool: %v", err)
+		klog.Error(err, "unable to get InferencePool")
 		return ctrl.Result{}, err
 	}
 
-	klog.V(2).Infof("Updated LLMServerPool: %+v", serverPool)
-	c.Datastore.setLLMServerPool(serverPool)
+	c.updateDatastore(serverPool)
 
 	return ctrl.Result{}, nil
 }
 
-func (c *LLMServerPoolReconciler) SetupWithManager(mgr ctrl.Manager) error {
+func (c *InferencePoolReconciler) updateDatastore(serverPool *v1alpha1.InferencePool) {
+	if c.Datastore.InferencePool == nil || serverPool.ObjectMeta.ResourceVersion != c.Datastore.InferencePool.ObjectMeta.ResourceVersion {
```

Comment:
I don't think this is necessary. I don't think we get here often if the resourceVersion didn't change. Even if we do, `setInferencePool` is very lightweight so calling it a few more times is fine.

Reply:
Noted. 

This is equally lightweight and it's expected for the pool config to be rather stable, so this avoids unnecessary locking, regardless of how infrequent or quick.